forward_engineering/ddlProvider/ddlHelpers/sequenceHelper.js

```diff
@@ -0,0 +1,172 @@
+ * increment?: number;
+ * maxValue?: number;
+ * minValue?: number;
+ * ownedByColumn: object[];
+ * ownedByNone: boolean;
+ * sequenceName: string;
+ * start?: number;
+ * temporary: boolean;
+ * unlogged: boolean;
+ * }} Sequence
+ * 
+ * @typedef {{
+ * key: keyof Sequence;
+ * clause: string;
+ * getOption: (sequence: Sequence, config: OptionConfig)=> {}
+ * }} OptionConfig
+ */
+
+module.exports = ({
+	_,
+	templates,
+	assignTemplates,
+	getNamePrefixedWithSchemaName,
+}) => {
+	/**
+	 * @param {string} schemaName 
+	 * @param {Sequence[]} sequences 
+	 * @returns {string}
+	 */
+	const getSequencesScript = (schemaName, sequences) => {
+		return _.map(sequences, (sequence) => {
+			const name = getNamePrefixedWithSchemaName(
+				sequence.sequenceName,
+				schemaName
+			);
+			const ifNotExists = getIfNotExists(sequence);
+			const sequenceType = getSequenceType(sequence);
+			const options = getSequenceOptions(sequence);
+			return assignTemplates(templates.createSequence, {
+				name,
+				ifNotExists,
+				sequenceType,
+				options,
+			});
+		}).join('\n');
+	};
+
+	/**
+	 * @param {Sequence} sequence 
+	 * @returns {string}
+	 */
+	const getSequenceOptions = (sequence) => {
+		const optionConfigs = [
+			{ getOption, key: 'dataType', clause: 'AS', },
+			{ getOption, key: 'increment', clause: 'INCREMENT', },
+			{ getOption, key: 'start', clause: 'START WITH', },
+			{ getOption, key: 'minValue', clause: 'MINVALUE', },
+			{ getOption, key: 'maxValue', clause: 'MAXVALUE', },
+			{ getOption, key: 'cache', clause: 'CACHE', },
+			{ getOption: getCycle, key: 'cycle' },
+			{ getOption: getOwnedBy, key: 'ownedByColumn' },
+		];
+
+		const options = optionConfigs
+			.map((config) => wrapOption(config.getOption(sequence, config)))
+			.filter(Boolean)
+			.join('');
+
+		return options ? wrapOptionsBlock(options) : options;
+	};
+
+	/**
+	 * @param {Sequence} sequence 
+	 * @param {OptionConfig} config 
+	 * @returns {string}
+	 */
+	const getOption = (sequence, config) => {
+		const value = sequence[config.key];
+		return value || value === 0 ? `${config.clause} ${value}` : '';
+	};
+
+	/**
+	 * @param {string} option 
+	 * @returns {string}
+	 */
+	const wrapOption = (option) => {
+		return option ? `\t${option}\n` : '';
+	};
+
+	/**
+	 * @param {string} option 
+	 * @returns {string}
+	 */
+	const wrapOptionsBlock = (option) => {
+		return '\n' + option.replace(/\n$/, '');
+	};
+
+	/**
+	 * @param {Sequence} sequence 
+	 * @returns {string}
+	 */
+	const getIfNotExists = (sequence) => {
+		return sequence.ifNotExist ? ' IF NOT EXISTS' : '';
+	};
+
+	/**
+	 * @param {Sequence} sequence 
+	 * @returns {string}
+	 */
+	const getSequenceType = (sequence) => {
+		if (sequence.temporary) {
+			return ' TEMPORARY';
+		}
+
+		if (sequence.unlogged) {
+			return ' UNLOGGED';
+		}
+
+		return '';
+	};
+
+	/**
+	 * @param {Sequence} sequence 
+	 * @returns {string}
+	 */
+	const getCycle = (sequence) => {
+		if (sequence.cycle === true) {
```

Comment:
> I think we can trust our [own code](https://github.com/hackolade/studio/pull/848/files#diff-a723c116d51b00a912946e8304d0275a4143821ca1a033311e5d4a147033168cR608) and simplify the check:

You are referring to code that relates to RE, and this PR is part of the FE logic : )
`cycle` can have `undefined` value, and since this clause is optional, we don't have to define it explicitly